### PR TITLE
ch setup continuous intregration

### PR DESCRIPTION
**Pull request by habaumugisha1**
# Description

This is for configuring continuous integration such as setting up the test environment on local and running on Travis, coveralls, code climate analysis, and maintainability

Fixes # (issue)
>No issue fixed because it a new feature
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

>This will be tested by running `npm test`

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/tests/app.test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { expect, request, use } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../app';
+
+use(chaiHttp);
+describe('testing welcome router', () => {
+  it('Should get welcome message', async () => {
+    const res = await request(app).get('/');
+    expect(res).to.have.status([200]);
+    expect('Welcome to Barefoot Nomad').to.equal('Welcome to Barefoot Nomad');
```
Comment (salviosage) [inline src/tests/app.test.js]: please change this  to `expect(res.text).to.equal('Welcome to Barefoot Nomad');`

File: package.json [hunk 2]
```diff
@@ -4,7 +4,10 @@
   "description": "Backend API for Barefoot Nomad ",
   "main": "app.js",
   "scripts": {
-    "start": "nodemon --exec babel-node app.js",
+    "start": "node_modules/.bin/babel-node ./app.js",
+    "cohort2-barefoot": "nodemon --exec babel-node app.js",
+    "test": "nyc --reporter=html --reporter=text mocha --require @babel/register ./src/tests/*.test.js --timeout 1000000 --exit",
```
Comment (salviosage) [inline package.json]: you should change environment to test too

**Review by salviosage — CHANGES_REQUESTED** [review]
Nice work, please attend my review

File: .coveralls.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+repo_token: LBqOgmrFuiVAaGMil54sD40Axr58PJ7kz
```
Comment (salviosage) [inline .coveralls.yml]: this token should come from `.env` file

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+language: node_js
+
+node_js:
+  - "stable"
+
```
Comment (salviosage) [inline .travis.yml:9]: add this `env:
  global:
    - NODE_ENV=test` learn more here [Here](https://docs.travis-ci.com/user/environment-variables/#global-variables)

**Review by salviosage — CHANGES_REQUESTED** [review]
Nice work ,I left some comments

**Review by inezabonte — CHANGES_REQUESTED** [review]
Great work. In the`welcome.js` controller you can change the response message to a `json` object to make it easier for testing
```javascript
res.status(200).json({message: 'Welcome to Barefoot Nomad'});
```

**Review by inezabonte — APPROVED** [review]

File: .env [hunk 5]
```diff
@@ -0,0 +1 @@
+COVERALLS_REPO_TOKEN=LBqOgmrFuiVAaGMil54sD40Axr58PJ7kz
```
Comment (salviosage) [inline .env]: Please delete this

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,9 @@
 node_modules
 .DS_Store
+
```
Comment (salviosage) [inline .gitignore:20]: you can add .env here

File: package.json [hunk 7]
```diff
@@ -4,7 +4,10 @@
   "description": "Backend API for Barefoot Nomad ",
   "main": "app.js",
   "scripts": {
-    "start": "nodemon --exec babel-node app.js",
+    "start": "node_modules/.bin/babel-node ./app.js",
+    "cohort2-barefoot": "nodemon --exec babel-node app.js",
```
Comment (salviosage) [inline package.json]: this might be too big instead of `ohort2-barefoot` you can say `dev`

File: src/controllers/welcome.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 // Welcome page controller
 
 const welcome = (req, res) => {
-  res.status(200).send('Welcome to Barefoot Nomad');
+  res.status(200).json({ status: 200, message: 'Welcome to Barefoot Nomad' });
```
Comment (salviosage) [inline src/controllers/welcome.js:4]: nice work here

**Review by salviosage — APPROVED** [review]
LGTM

**Review by inezabonte — APPROVED** [review]

**Pull request merged**